libraries/AP_Param/AP_Param.h

```diff
@@ -895,9 +855,7 @@ class AP_ParamT : public AP_Param
 
     /// AP_ParamT types can implement AP_Param::cast_to_float
     ///
-    float cast_to_float(void) const {
-        return (float)_value;
-    }
+    float cast_to_float(void) const;
```

Comment:
this should stay as an inline IMO, I don't think this will save any flash

Reply:
Interestingly, this move saves 320 bytes on CubeOrange but costs on 40 on periph. I guess its better to save a little on the small boards than a lot on the big boards. I will wait for the CI size test to complete before pushing as we have a better comparison.

Reply:
As it is now:

Periph:
```
Binary Name    Text [B]        Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
-------------  --------------  -----------  -------------  ----------------------------  -------------------------
ap_periph      220 (+0.2190%)  0 (0.0000%)  -4 (-0.0199%)  220 (+0.2186%)                                     4616
```
Pixhawk 1- 1M
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
antennatracker   -540 (-0.0720%)   0 (0.0000%)  4 (+0.0020%)   -540 (-0.0719%)                                  281816
arducopter-heli  -1096 (-0.1084%)  0 (0.0000%)  0 (0.0000%)    -1096 (-0.1083%)                                  21124
ardurover        -1168 (-0.1274%)  0 (0.0000%)  0 (0.0000%)    -1168 (-0.1273%)                                 115556
arducopter       -1268 (-0.1269%)  0 (0.0000%)  -4 (-0.0020%)  -1268 (-0.1268%)                                  33532
ardusub          -1160 (-0.1263%)  0 (0.0000%)  0 (0.0000%)    -1160 (-0.1262%)                                 114040
blimp            -444 (-0.0627%)   0 (0.0000%)  -4 (-0.0020%)  -444 (-0.0627%)                                  324124
arduplane        -1304 (-0.1267%)  0 (0.0000%)  0 (0.0000%)    -1304 (-0.1266%)                                   3284
```
F405
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
antennatracker   -620 (-0.0945%)   0 (0.0000%)  4 (+0.0031%)   -620 (-0.0944%)                                  326796
arducopter-heli  -1312 (-0.1388%)  0 (0.0000%)  0 (0.0000%)    -1312 (-0.1385%)                                  36972
ardurover        -1376 (-0.1613%)  0 (0.0000%)  0 (0.0000%)    -1376 (-0.1612%)                                 130692
arducopter       -1484 (-0.1587%)  0 (0.0000%)  -4 (-0.0031%)  -1484 (-0.1583%)                                  47244
ardusub          -1348 (-0.1656%)  0 (0.0000%)  -4 (-0.0031%)  -1348 (-0.1655%)                                 169736
blimp            -596 (-0.0883%)   0 (0.0000%)  4 (+0.0031%)   -596 (-0.0882%)                                  307552
arduplane        -1556 (-0.1596%)  0 (0.0000%)  4 (+0.0031%)   -1556 (-0.1593%)                                   7516
```
Durandal
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
antennatracker   -3256 (-0.2406%)  0 (0.0000%)  0 (0.0000%)    -3256 (-0.2402%)                                 613972
arducopter-heli  -4848 (-0.2714%)  0 (0.0000%)  0 (0.0000%)    -4848 (-0.2709%)                                 181548
ardurover        -4360 (-0.2719%)  0 (0.0000%)  0 (0.0000%)    -4360 (-0.2716%)                                 364876
arducopter       -5236 (-0.2944%)  0 (0.0000%)  4 (+0.0015%)   -5236 (-0.2938%)                                 189372
ardusub          -4596 (-0.2937%)  0 (0.0000%)  -4 (-0.0015%)  -4596 (-0.2934%)                                 404104
blimp            -3240 (-0.2500%)  0 (0.0000%)  0 (0.0000%)    -3240 (-0.2497%)                                 671712
arduplane        -5172 (-0.2936%)  0 (0.0000%)  4 (+0.0015%)   -5172 (-0.29%)                                 206708
```

Reply:
So, adding back this method has saved 32 bytes on periph, 40 bytes on Pixhawk1 and F405 but cost us 300 bytes on Durandal.